analyzers/src/SonarAnalyzer.VisualBasic/Rules/RouteTemplateShouldNotStartWithSlash.cs

```diff
@@ -0,0 +1,31 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.VisualBasic;
+
+[DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+public sealed class RouteTemplateShouldNotStartWithSlash : RouteTemplateShouldNotStartWithSlashBase<SyntaxKind>
+{
+    protected override ILanguageFacade<SyntaxKind> Language => VisualBasicFacade.Instance;
+    protected override SyntaxKind MethodSyntaxKind => SyntaxKind.FunctionBlock;
+
+    protected override Location ControllerLocation(INamedTypeSymbol symbol) =>
+        ((ClassStatementSyntax)symbol.GetFirstSyntaxRef()).GetIdentifier()?.GetLocation();
```

Comment:
is there a more elegant way to do this cast?
I'm not checking for null as we know it's a class declaration for sure.
Also in case this is a partial class it will pick one of the two references - I should deal with it in another PR as well.

Reply:
That is the correct way, but two adoptions are needed (the second being optional).
You need the first SyntaxRef, that is not in generated code.
Also, there may be more than one location that is not in generated code. These should be secondary locations as well, but I think this part is not that important

Reply:
> Also, there may be more than one location that is not in generated code. These should be secondary locations as well, but I think this part is not that important

I'll deal with it in the partial class part of the initial issue.